### PR TITLE
Capture flow completed

### PR DESCRIPTION
**Pull request by EngrTaofeek**
I have completed the capture flow as indicated in the issue assigned to me. The items now saves to the database and the back button takes the user to the dashboard.

### CODE REVIEW

File: app/src/main/java/org/greenstand/android/TreeTracker/usecases/CreateTreeUseCase.kt [hunk 1]
```diff
@@ -34,10 +34,11 @@ class CreateTreeUseCase(
...
         )
         analytics.treePlanted()
-        val attributeEntitites = params.treeCaptureAttributes().map {
-            TreeAttributeEntity(it.key, it.value, -1)
-        }.toList()
-        Timber.d("Inserting TreeCapture entity $entity")
+        val attributeEntitites: List<TreeAttributeEntity>? = params.treeCaptureAttributes().entries.map {
+            TreeAttributeEntity("${it.key}", "${it.value}" , -1)
+        }
+
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/usecases/CreateTreeUseCase.kt:40]: What was this part changed for?

File: app/src/main/java/org/greenstand/android/TreeTracker/capture/TreeCaptureScreen.kt [hunk 2]
```diff
@@ -48,7 +48,7 @@ fun TreeCaptureScreen(
             ActionBar(
                 leftAction = {
                     ArrowButton(isLeft = true) {
-                        navController.popBackStack()
+                        navController.navigate(NavRoute.Dashboard.route)
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/capture/TreeCaptureScreen.kt:51]: This navigates back to the dashboard, but if I press 'back' on the dashboard I go to the camera screen again.

File: app/src/main/java/org/greenstand/android/TreeTracker/camera/Camera.kt [hunk 3]
```diff
@@ -75,6 +75,7 @@ fun Camera(
                         )
                         .build()
 
+                    cameraProvider.unbindAll()
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/camera/Camera.kt:78]: This seems like it does a reset right before we load the camera. Would it work better if we unbindAll right after taking the picture?

**Review by Elforama — CHANGES_REQUESTED** [review]
Overall, it's looking good. Just some things to address.

File: app/src/main/java/org/greenstand/android/TreeTracker/usecases/CreateTreeUseCase.kt (hunk 1 above)
Comment (EngrTaofeek) [reply app/src/main/java/org/greenstand/android/TreeTracker/usecases/CreateTreeUseCase.kt]: It was actually not necessary, I have replaced it with the existing one.

File: app/src/main/java/org/greenstand/android/TreeTracker/capture/TreeCaptureScreen.kt (hunk 2 above)
Comment (EngrTaofeek) [reply app/src/main/java/org/greenstand/android/TreeTracker/capture/TreeCaptureScreen.kt]: Okay fixed this in a new commit.

File: app/src/main/java/org/greenstand/android/TreeTracker/camera/Camera.kt (hunk 3 above)
Comment (EngrTaofeek) [reply app/src/main/java/org/greenstand/android/TreeTracker/camera/Camera.kt:78]: I was getting an error that I was trying to bind too many usecases. So it wasn't able to take/save the picture and this fixed it.

**Review by Elforama — APPROVED** [review]
Great work, this looks good!

**Pull request merged**